addons/stock/models/stock_quant.py

```diff
@@ -134,8 +134,9 @@ def _search_on_hand(self, operator, value):
         """Handle the "on_hand" filter, indirectly calling `_get_domain_locations`."""
         if operator not in ['=', '!='] or not isinstance(value, bool):
             raise UserError(_('Operation not supported'))
-        domain_loc = self.env['product.product']._get_domain_locations()[0]
-        quant_ids = [l['id'] for l in self.env['stock.quant'].search_read(domain_loc, ['id'])]
+        domain_loc = self.env['product.product'].with_context(compute_child=False)._get_domain_locations()[0]
+        location_ids = self.env['stock.location'].search([('id', 'child_of', domain_loc[0][2])]).ids
+        quant_ids = self.env['stock.quant'].search([('location_id', 'in', location_ids)]).ids
```

Comment:
We should use _search (return a subquery instead of execute it) to avoid to fetch all quants_ids in the python side. (can be done also for the previous search). Can you try ?

Reply:
@ryv-odoo Yep, now the search_read takes between 40ms-100ms. I didn't know about this `_search` trick 😮